### ISSUE TITLE
I downloaded UGminer and it does not start downloading the individual miners.

### ISSUE THREAD

**Issue by MotikaParva** [UselessGuru/UG-Miner#1]
I start the bat, the code runs gets the available pools, but it does not start downloading the actual miners. I can download them myself and save them to a folder, but I am certain it is not how it is supposed to work. 😃 

Are you still working on this project @UselessGuru?

**Comment by UselessGuru** [comment]
> Are you still working on this project

Yes. I was just a away for a while.

> I am certain it is not how it is supposed to work. 

You are right. It should automatically download all required binaries - which it does here :-)

To troubleshoot I recommend you enable debug logging in the configuration editor:
<img width="248" alt="image" src="https://github.com/UselessGuru/UG-Miner/assets/30080938/1e341cba-b9cc-46df-8061-cd48c52a5018">

Do you get any messages like 
2024-06-08 09:28:47 INFO: Downloader: Initiated download of 'https://github.com/develsoftware/GMinerRelease/releases/download/3.44/gminer_3_44_windows64.zip'.
???

Are you running UG-Miner as admin? See if runing as local admin helps. I am interested to find out what is wrong in your case.

**Comment by MotikaParva** [comment]
When I run the UG-miner.bat file by double clicking it runs and it gets stuck at:

> 2024-06-10 12:12:34 DEBUG: Brain 'ZergPool': End loop (Duration 6.7678968 sec. / Avg. loop duration: 6.7678968 sec.); Price history 181 objects; found 181 valid pools.
It does not start downloading any miners at all.

When I right click and Run as Administrator it opens PowerShell and it closes immediately.
I added '-NoExit' to the .bat file so I can see why it errors.

I got this:
![Screenshot 2024-06-10 182033](https://github.com/UselessGuru/UG-Miner/assets/105045721/2e344528-f6fd-472b-a59c-a066e8e0a6e7)
I updated PowerShell to the latest version 7.4.2, but I still get the same error. I am sure its something I miss.
Just to test things out I downloaded last version of Nemos and ran it without issues. Both miner.bat and miner_AsAdmin.bat run normally.
Any ideas would be greatly appreciated!

**Comment by UselessGuru** [comment]
> I updated PowerShell to the latest version 7.4.2

I recommend NOT to use 7.4.2 - it has a memory leak. It is better to stick with 7.2.19.

> When I right click and Run as Administrator it opens PowerShell and it closes immediately.

I have seen this before, but it is not caused by the UG-Miner code itself, much more i believe it is a local config issue (path set correctly?). Try uncommenting the first line (`@echo off` -> `#@echo` off) in UG-Miner_asAdmin.bat. So you can see each commend as it gets executed. Please upload a complete screen shot - thank you!

As a workaround to find the cause of the non-download please try this:

Open a CMD window as admin. Then change to the UG-Miner directory. Now run 'UG-Miner.cmd'.

> 2024-06-10 12:12:34 DEBUG: Brain 'ZergPool': End loop (Duration 6.7678968 sec. / Avg. loop duration: 6.7678968 sec.); Price > history 181 objects; found 181 valid pools.
> It does not start downloading any miners at all.

What is the next line AFTER `'found 181 valid pools.'`?

**Comment by MotikaParva** [comment]
> I recommend NOT to use 7.4.2 - it has a memory leak. It is better to stick with 7.2.19.

OK the first thing I did is I reverted back to 7.2.19. Just wanted to make sure its not something weird due to not being up to date and this is why I tested 7.4.2...

> What is the next line AFTER 'found 181 valid pools.'?

There is nothing afterwards. It just stops there. I waited 5-10 min just to make sure, but nothing is happening.

Now I downloaded UG-Miner - 6.2.9
My first attempt without changing anything produced the same result. I open UG-Miner.bat as Admin and it opens PowerShell window for 1 or 2 seconds and immediately closes it.

Adding the `#@echo off` and `-noexit`:

![Screenshot 2024-06-15 102646](https://github.com/UselessGuru/UG-Miner/assets/105045721/e808eabc-7cce-4ba3-a650-cc6b92565449)

> As a workaround to find the cause of the non-download please try this:
> Open a CMD window as admin. Then change to the UG-Miner directory. Now run 'UG-Miner.cmd'.

I tried the workaround you recommended. The `UG-Miner.cmd` did not work, but I tested `UG-Miner.bat` and it ran:

![Screenshot 2024-06-15 104258](https://github.com/UselessGuru/UG-Miner/assets/105045721/dc3be11b-8f9b-4281-b221-42d578c158f2)
![Screenshot 2024-06-15 104517](https://github.com/UselessGuru/UG-Miner/assets/105045721/1087e644-f7e4-4ca9-9006-3020f34dd7b8)

I waited about 5 minutes again, but to me it looks like it just stops there again.
Instead of sending screenshots of SnakeTrail I am adding this file:

[UG-Miner_2024-06-15.log](https://github.com/user-attachments/files/15846128/UG-Miner_2024-06-15.log)

Any help would be greatly appreciated. Meanwhile I will reinstall [Visual C+ RunTimes](https://github.com/UselessGuru/Visual-C-Runtimes-All-in-One-Sep-2019/releases/download/sep2019/Visual-C-Runtimes-All-in-One-Sep-2019.zip) and [Windows Nvidia driver 537.42](https://us.download.nvidia.com/Windows/537.42/537.42-desktop-win10-win11-64bit-international-dch-whql.exe) just in case.
Thank for your time!

**Comment by MotikaParva** [comment]
Just FYI when I tried downloading the [Visual C+ RunTimes](https://github.com/UselessGuru/Visual-C-Runtimes-All-in-One-Sep-2019/releases/download/sep2019/Visual-C-Runtimes-All-in-One-Sep-2019.zip) it says `Not Found` for some reason.

**Comment by UselessGuru** [comment]
I think I have ideas for the causes of your issues:
a) UG-Miner not starting
Windows 11 defaults to use WindowsTerminal for PWSH sessions. This is causing issues with UG-Miner (I am investigating).
As a workaround try:
Edit UG-Miner.bat
and change
`PWSH -executionpolicy bypass -windowstyle maximized -command ".\UG-Miner.ps1 -ConfigFile '.\Config\config.json'"`
to
`PWSH -executionpolicy bypass -windowstyle maximized -command "%~dp0UG-Miner.ps1" -ConfigFile "%~dp0Config\config.json"`

You can also try to set 'Windows Console Host' as default in Windows Terminal settings.

b) UG-Miner stopping
Have you saved your config in the configuration editor? It needs applying 'Save config' at least once (a fix for this is in the works)

**Comment by UselessGuru** [comment]
> Just FYI when I tried downloading the [Visual C+ RunTimes](https://github.com/UselessGuru/Visual-C-Runtimes-All-in-One-Sep-2019/releases/download/sep2019/Visual-C-Runtimes-All-in-One-Sep-2019.zip) it says Not Found for some reason.

Try this:
https://github.com/UselessGuru/UG-Miner-Extras/releases/download/Visual-C-Runtimes-All-in-One-Sep-2019/Visual-C-Runtimes-All-in-One-Sep-2019.zip

The link will be updated in the next version

**Comment by MotikaParva** [comment]
> As a workaround try:
Edit UG-Miner.bat
and change
`PWSH -executionpolicy bypass -windowstyle maximized -command ".\UG-Miner.ps1 -ConfigFile '.\Config\config.json'"`
to
`PWSH -executionpolicy bypass -windowstyle maximized -command "%~dp0UG-Miner.ps1" -ConfigFile "%~dp0Config\config.json"`

This helped. It gave me an error it does not recognize the name of the folder as a command. I am assuming it was the space in the naming of the folder. 
I renamed the folder to UGM and it is now downloading the binaries. 💪
After it is done downloading and benchmarks them all I will report back. Thanks a lot! 

> Have you saved your config in the configuration editor? It needs applying 'Save config' at least once (a fix for this is in the works)

I did. It would not let me `Start Mining` without setting `Main currency` and afterwards I need to `Save Configuration`.

**Comment by UselessGuru** [comment]
> I did. It would not let me Start Mining without setting Main currency

That is intentional. Main currency is mandatory.
Wasn't there your local currency already filled in?

**Comment by MotikaParva** [comment]
When I try to start the UG-Miner_AsAdmin.bat I am getting the following error since the last update. FYI
![Screenshot 2024-06-30 200536](https://github.com/UselessGuru/UG-Miner/assets/105045721/ab911925-da8a-449a-89e5-a646943a139b)

**Comment by UselessGuru** [comment]
Where did you get 6.2.9 from? Current version is 6.2.13 (released today).

The issue you are seeing is actually caused by a very old windows problem. Internally a lot of commands still rely on the old 8.3 file naming requirement. This is not a UG-Miner issue per se. Unfortunately the only was to get around this problem is to only use file and folder names matching the old 8.3 naming scheme or to disable 8.3 filenames. For more infomation see here:
https://schneegans.de/windows/no-8.3/

I suggest you rename your 'UGM' folder to something else. Then create 'UGM' folder again (do not use a folder with version number - this would no longer match after an automatic update), install the latest version 6.2.13 directly to this folder. You can copy the 'Config', 'Logs' and 'Stats' folder from the renamed folder.

**Comment by MotikaParva** [comment]
> Where did you get 6.2.9 from? Current version is 6.2.13 (released today).

I had the idea of keeping different versions in different folders just in case. I am seeing that auto update makes this redundant and unnecessary. This is why I still have the 6.2.9 name on the folder, but the contents are up to date.

> I suggest you rename your 'UGM' folder to something else.

I now realize I wont need any more sub directories in that UGM folder and will keep everything there. Thanks for the help.

I am seeing a few miners with issues (mining not recording stats, not mining, etc). I am waiting on getting a full list to post 
them as one message not miner by miner.
Should I create a new issue or list them here?

**Comment by MotikaParva** [comment]
Just FYI. Installing a fresh copy of 6.2.13 produces the same error.

I just downloaded a fresh 6.2.12 and it ran with no issues. It auto updated to 6.2.13 and I then I saw the same error.
It may be an issue with my machine, but it shows after I update to 6.2.13.

**Comment by UselessGuru** [comment]
> Should I create a new issue or list them here?

List them here. I will see what I can do. I do not have the same hardware as you so this will certainly help to exclude non-working miners. Sometimes the failure may also be related to pool issues. E.g. ZergPool frequently has issues with SSL connections.

> but it shows after I update to 6.2.13.

Very strange. I have 3 machines here and none of them show this issue.

Can you try removing all 8.3 filenames in UG-Miner directory? Use this command:
`fsutil 8dot3name strip [UG-Miner Directory] /s`
e.g. fsutil 8dot3name strip C:\Users\Owner\Desktop\UGM /s

**Comment by UselessGuru** [comment]
@MotikaParva

Any update?
